runtime/bcutil/ClassFileWriter.cpp

```diff
@@ -493,6 +498,37 @@ ClassFileWriter::analyzeMethods()
 	}
 }
 
+void
+ClassFileWriter::analyzeRecordAttribute()
+{
+	addEntry((void *) &RECORD, 0, CFR_CONSTANT_Utf8);
+
+	/* first 4 bytes contains number of record components */
+	U_32 numberOfRecords = getNumberOfRecordComponents(_romClass);
+	J9ROMRecordComponentShape* recordComponent = recordComponentStartDo(_romClass);
+	for (U_32 i = 0; i < numberOfRecords; i++) {
+
+		/* record component name and signature */
+		addEntry(J9ROMRECORDCOMPONENTSHAPE_NAME(recordComponent), 0, CFR_CONSTANT_Utf8);
+		addEntry(J9ROMRECORDCOMPONENTSHAPE_SIGNATURE(recordComponent), 0, CFR_CONSTANT_Utf8);
+
+		/* analyze attributes */
+		if (recordComponentHasSignature(recordComponent)) {
+			J9UTF8* genericSignature = getRecordComponentGenericSignature(recordComponent);
+			addEntry((void *) &SIGNATURE, 0, CFR_CONSTANT_Utf8);
+			addEntry(genericSignature, 0, CFR_CONSTANT_Utf8);
+		}
+		if (recordComponentHasAnnotations(recordComponent)) {
+			addEntry((void *) &RUNTIME_VISIBLE_ANNOTATIONS, 0, CFR_CONSTANT_Utf8);
+		}
+		if (recordComponentHasTypeAnnotations(recordComponent)) {
+			addEntry((void *) &RUNTIME_VISIBLE_TYPE_ANNOTATIONS, 0, CFR_CONSTANT_Utf8);
```

Comment:
Do the typeAnnotations need to be analyzed here as well?  Same for the component annotations?   Otherwise I'm not sure how their UTF8s get added

Reply:
Looking deeper into the code, you've matched the existing pattern and I think I understand how this works now.